ecs.php

```diff
@@ -40,6 +40,9 @@
     // make ECS 16x faster
     $parameters->set(Option::PARALLEL, true);
 
+    // Reusable ECS cache for Docker runtime
+    $parameters->set(Option::CACHE_DIRECTORY, 'tmp/ecs');
+
     $parameters->set(Option::PATHS, [
```

Comment:
The configs cannot be changed to make Docker happy, as the default way to develop Rector is natively. Same for PHPStan.

Reply:
But why? Default values make use of local system temp folder and it's not possible to use cache inside Docker container (in fact, temp dir is used inside container but it disappears when container stops, so cache lives only through one `run` lifecycle and can't be reused). I don't see how it collides with non-Docker development 🤔 You'll just have cache in project's dir, not somewhere else in the filesystem.

Reply:
It's our convention to narrow the debug use cases :). 
We used to follow different path before, adding more Docker-features to core for occasional contributors, but nobody used them and knew how to work with them. To make our lives easier and development more reliable, we've decide to follow one standard.

Reply:
Honestly, I don't understand this reasoning. For non-Docker usage it does not make difference where cache is stored, but for Docker it is really important because cache will work or not 🙂 Without specifying cache paths it becomes really hard to use because every run is like first run and takes much time. I believe nobody will even notice that cache moved to different directory, this change is 100% transparent for developers because nothing needs to be done in order to work. It just enables reusable cache inside Docker container.

Reply:
It does matter actually, as clearing cache is related to this path etc.

Feel free to move it to Docker setup scripts :)

Reply:
So clearing cache should be improved to be compatible 😉 Please point me to scripts/code that stopped working after this change, because I don't see any actual impact. I just don't get what's the problem with explicit path to cache, using already existing `tmp` dir. I just saw that `tmp/rector` is already there and used the same directory.

It's impossible to move it to "Docker setup scripts" since there are not any wrapper scripts where additional options could be added, also ECS and PHPStan does not have option for setting cache path on CLI level. The only way that could work would be adding some ugly workarounds for copying/extending configs in real time.

Reply:
@TomasVotruba ping, can you provide answer for my concerns above? Because I don't know if we can find some way to keep this explicit cache path and without that it does not make sense to run these tools inside container (full analysis every time).

Reply:
Cache directories can be improved. The issue I have is with leaking this knowledge to all the configs we use :).
This should be handled on cache level itself.

Reply:
This was not resolved. I don't know the solution, but it can't be here :)

Reply:
@TomasVotruba there's nothing to resolve IMHO, this change does not break anything and works for native usage too (all checks green!). It just ensures the common, stable bridge between local filesystem and mounted volume.